### PR TITLE
Commit for features of Home Task 1, including different component types

### PR DESCRIPTION
**Pull request by pconnell99**
Built out three components for three widgets of the project, one function, two class.  Started styling the CSS a bit.

### CODE REVIEW

File: src/GenreToggle.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        };
+    }
+
+    handleChange = genre => {
+        this.setState({ chosenGenre: genre });
+        this.props.onGenreSelected(genre);
+    }
+
+    genreItems = () => {
+        return this.props.genres.map(genre => {
```
Comment (Dmitry-White) [inline src/GenreToggle.jsx:21]: A couple of tips to improve readability and have a more robust JSX:
1. If the only thing you have in your function is a return statement - use short version
`genreItems = () => this.props.genres.map(...);`
2. Extract JS out of JSX, especially conditional statements
`const effectiveClassName =  this.state.chosenGenre === genre ? "genreItem selected" : "genreItem";`
3. When passing callbacks/arrow function as props - use short version
`onClick={() => this.handleChange(genre)}`

**Review by Dmitry-White — CHANGES_REQUESTED** [review]
Looks great!
Left a couple of comments for further improvements ;)

File: src/App.js [hunk 2]
```diff
@@ -1,23 +1,20 @@
-import logo from './logo.svg';
 import './App.css';
+import { IncrementValue } from './IncrementValue'
+import React from 'react';
+import SearchBar from './SearchBar';
+import GenreToggle from './GenreToggle';
 
 function App() {
+  let genres = ["All", "Documentary", "Comedy", "Horror", "Crime"];
```
[the remark this answers is not in the document]
Comment (pconnell99) [reply src/App.js:8]: My bad.  As I am the sole dev normally, I tend to dev 'knowing' that it is a constant.  Agreed, will remember.

File: src/App.js [hunk 3]
```diff
@@ -1,23 +1,20 @@
...
-          Learn React
-        </a>
-      </header>
+      {React.createElement("div", {}, "Hello World!")}
+      <br />
+      <IncrementValue />
+      <br />
+      <SearchBar />
+      <br />
+      <GenreToggle genres={genres} onGenreSelected={(genre) => { alert(genre) }} />
```
[the remark this answers is not in the document]
Comment (pconnell99) [reply src/App.js:17]: Was purely for demonstration purposes of passing vars back up - would never use in production, agreed.

File: src/GenreToggle.jsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
+/** Genre Toggle Component 
+ * Using PureComponent to stop renrendering on state in parent changing potentially (https://betterprogramming.pub/when-to-use-react-purecomponent-723f85738be1)
+*/
+
+import React from "react";
+
+export default class GenreToggle extends React.PureComponent {
+    constructor(props) {
```
[the remark this answers is not in the document]
Comment (pconnell99) [reply src/GenreToggle.jsx:8]: Ok.

File: src/GenreToggle.jsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+            chosenGenre: "All" // default
+        };
+    }
+
+    handleChange = genre => {
+        this.setState({ chosenGenre: genre });
+        this.props.onGenreSelected(genre);
+    }
+
+    genreItems = () => {
```
[the remark this answers is not in the document]
Comment (pconnell99) [reply src/GenreToggle.jsx:20]: Agreed, I want as little JS manip. in render as I can get away with, MAYBE some ? in there, but that's about it.

File: src/GenreToggle.jsx (hunk 1 above)
Comment (pconnell99) [reply src/GenreToggle.jsx:21]: 2. - Agreed, useful optimisation.
1 and 3. - Personal preference.  I find it more 'readable' when scanning code to have unambiguous scopes defined, even on single returns - but I'll of course follow any style guide available.

**Pull request merged**